actix-identity/src/identity_errors.rs

```diff
@@ -0,0 +1,91 @@
+use actix_session::{SessionGetError, SessionInsertError};
+use actix_web::{cookie::time::error::ComponentRange, http::StatusCode, ResponseError};
+
+/// This error can occur during login attempts.
+#[derive(Debug)]
+pub struct LoginError(SessionInsertError);
+
+impl std::fmt::Display for LoginError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl std::error::Error for LoginError {
+    fn source(&self) -> Option<&(dyn std::error::Error + 'static)> {
+        Some(&self.0)
+    }
+}
+
+impl ResponseError for LoginError {
+    fn status_code(&self) -> StatusCode {
+        StatusCode::UNAUTHORIZED
+    }
+}
+
+impl From<SessionInsertError> for LoginError {
+    fn from(error: SessionInsertError) -> Self {
+        Self(error)
+    }
+}
+
+/// This error describes all of the potential failures which can happen
+/// while retrieving an identity.
+#[derive(Debug)]
+#[non_exhaustive]
+pub enum GetIdentityError {
+    /// This is an error which shouldn't occur, and indicates some kind of bug.
+    LostIdentityError,
+    /// This occurs whenever no identity is found in a session.
+    MissingIdentityError,
+
+    /// This occurs whenever something goes wrong accessing a session store.
+    SessionGetError(SessionGetError),
+    /// This occurs whenever any kind of expiration of a session has taken place.
+    SessionExpiryError(ComponentRange),
```

Comment:
I'd prefer not to expose `ComponentRange` here - it'd be better to introduce a `SessionExpiryError` struct that wraps `ComponentRange` and doesn't expose it.

Reply:
Cool - I _think_ I've done enough to keep `ComponentRange` behind that struct, let me know if I missed anything there?

Reply:
It's looking good!